After upgrade to Graylog 4.x, the Streams page does not load (StreamDTO - Null matchingType)

<!--- Provide a general summary of the issue in the Title above -->

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->
The Streams page should load successfully.

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

  After upgrading to Graylog 4.x, the Stream page does not load. The following stacktrace is displayed in the Graylog server log:

```
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `org.graylog2.streams.StreamDTO`, problem: Null matchingType
at org.graylog2.streams.StreamDTO.create(StreamDTO.java:141) ~[graylog.jar:?]
at org.graylog2.streams.$AutoValue_StreamDTO.<init>($AutoValue_StreamDTO.java:71) ~[graylog.jar:?]
at org.graylog2.streams.AutoValue_StreamDTO.<init>(AutoValue_StreamDTO.java:19) ~[graylog.jar:?]
at org.graylog2.database.PaginatedDbService.findPaginatedWithQueryFilterAndSort(PaginatedDbService.java:159) ~[graylog.jar:?]
at org.graylog2.streams.PaginatedStreamService.findPaginated(PaginatedStreamService.java:48) ~[graylog.jar:?]
```

This indicates that the `streams.matching_type` collection value is null, but the `StreamDTO` requires a value.

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

Allow the [StreamDTO.matchingType()](https://github.com/Graylog2/graylog2-server/blob/29aa0ab4bd8dca7ef60ab7b5f1b17b805d9e7027/graylog2-server/src/main/java/org/graylog2/streams/StreamDTO.java#L66) to be Optional or nullable. Also, provide a default value in the create or builder methods. The default can use the same logic as [StreamImpl](https://github.com/Graylog2/graylog2-server/blob/29aa0ab4bd8dca7ef60ab7b5f1b17b805d9e7027/graylog2-server/src/main/java/org/graylog2/streams/StreamImpl.java#L223) uses currently.

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
This issue occurs mostly for older Graylog installations that are upgraded to 4.1 or beyond.

## Workaround
To workaround the issue, provide a `streams.matching_type` value equal to `AND` in MongoDB for any that currently have a null value.

See user suggestions here: 
https://github.com/Graylog2/graylog2-server/issues/9637#issuecomment-827091749
https://github.com/Graylog2/graylog2-server/issues/9637#issuecomment-769005192

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->

* Graylog Version: 4.x

--- a/graylog2-server/src/main/java/org/graylog2/streams/StreamDTO.java
+++ b/graylog2-server/src/main/java/org/graylog2/streams/StreamDTO.java
@@ -19,23 +19,22 @@ package org.graylog2.streams;
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.google.auto.value.AutoValue;
 import org.bson.types.ObjectId;
 import org.graylog.autovalue.WithBeanGetter;
+import org.graylog2.plugin.streams.Stream;
 import org.graylog2.plugin.streams.StreamRule;
 import org.graylog2.rest.models.alarmcallbacks.requests.AlertReceivers;
 import org.graylog2.rest.models.streams.alerts.AlertConditionSummary;
-import org.graylog2.rest.models.system.outputs.responses.OutputSummary;
 
 import javax.annotation.Nullable;
 import java.util.Collection;
-import java.util.Collections;
-
-import static com.google.common.base.MoreObjects.firstNonNull;
 
 @AutoValue
 @WithBeanGetter
 @JsonAutoDetect
+@JsonDeserialize(builder = StreamDTO.Builder.class)
 public abstract class StreamDTO {
     public static final String FIELD_ID = "_id";
     public static final String FIELD_TITLE = "title";
@@ -108,45 +107,18 @@ public abstract class StreamDTO {
 
     public abstract Builder toBuilder();
 
-    @JsonCreator
-    public static StreamDTO create(@JsonProperty(FIELD_ID) String id,
-                                   @JsonProperty(FIELD_CREATOR_USER_ID) String creatorUserId,
-                                   @JsonProperty(FIELD_OUTPUTS) @Nullable Collection<ObjectId> outputs,
-                                   @JsonProperty(FIELD_MATCHING_TYPE) String matchingType,
-                                   @JsonProperty(FIELD_DESCRIPTION) @Nullable String description,
-                                   @JsonProperty(FIELD_CREATED_AT) String createdAt,
-                                   @JsonProperty(FIELD_DISABLED) boolean disabled,
-                                   @JsonProperty(FIELD_RULES) @Nullable Collection<StreamRule> rules,
-                                   @JsonProperty(EMBEDDED_ALERT_CONDITIONS) @Nullable Collection<AlertConditionSummary> alertConditions,
-                                   @JsonProperty(FIELD_ALERT_RECEIVERS) @Nullable AlertReceivers alertReceivers,
-                                   @JsonProperty(FIELD_TITLE) String title,
-                                   @JsonProperty(FIELD_CONTENT_PACK) @Nullable String contentPack,
-                                   @JsonProperty(FIELD_DEFAULT_STREAM) @Nullable Boolean isDefault,
-                                   @JsonProperty(FIELD_REMOVE_MATCHES_FROM_DEFAULT_STREAM) @Nullable Boolean removeMatchesFromDefaultStream,
-                                   @JsonProperty(FIELD_INDEX_SET_ID) String indexSetId) {
-        return new AutoValue_StreamDTO(
-                id,
-                creatorUserId,
-                outputs,
-                matchingType,
-                description,
-                createdAt,
-                rules,
-                disabled,
-                alertConditions,
-                alertReceivers,
-                title,
-                contentPack,
-                firstNonNull(isDefault, false),
-                firstNonNull(removeMatchesFromDefaultStream, false),
-                indexSetId);
+    static Builder builder() {
+        return Builder.create();
     }
 
     @AutoValue.Builder
     public abstract static class Builder {
         @JsonCreator
         public static Builder create() {
-            return new AutoValue_StreamDTO.Builder();
+            return new AutoValue_StreamDTO.Builder()
+                    .matchingType(Stream.MatchingType.AND.toString())
+                    .isDefault(false)
+                    .removeMatchesFromDefaultStream(false);
         }
 
         @JsonProperty(FIELD_ID)

